napari/layers/labels/labels.py

```diff
@@ -271,13 +272,41 @@ def contiguous(self, contiguous):
     @property
     def n_dimensional(self):
         """bool: paint and fill edits labels across all dimensions."""
-        return self._n_dimensional
+        warnings.warn(
+            trans._(
+                'Labels.n_dimensional is deprecated. '
+                'Use Labels.n_edit_dimensions instead.'
```

Comment:
Hi @jni I think we are adding a `deferred=True` to exceptions and warrnings.

Reply:
Oops! Actually those commits might predate that discussion, this has been on my backburner for a while! :sweat_smile: